pixeltable/globals.py

```diff
@@ -600,18 +601,32 @@ def create_dir(
         Create a directory and replace if it already exists:
 
         >>> pxt.create_dir('my_dir', if_exists='replace_force')
+
+        Create a subdirectory along with its ancestors:
+
+        >>> pxt.create_dir('parent1.parent2.sub_dir', create_parents=True)
     """
     path_obj = catalog.Path(path)
     cat = Catalog.get()
 
     with env.Env.get().begin_xact():
+        if create_parents:
+            # start walking from the root
+            ancestors = path_obj.ancestors()
+            root_path = catalog.Path('', empty_is_valid=True)
+            last_parent = cat.get_schema_object(str(root_path))
+            for ancestor in ancestors:
+                ancestor_obj = cat.get_schema_object(str(ancestor), expected=catalog.Dir)
+                last_parent = (
+                    catalog.Dir._create(last_parent._id, ancestor.name) if ancestor_obj is None else ancestor_obj
+                )
+            # get parent again
```

Comment:
Again? Also, isn't this last_parent?

Shouldn't there be an else branch?

Reply:
good catch we are not fetching parent at the beginning like earlier version of the code, will remove it and yes last parent should be the parent.